### PR TITLE
Return all project ids a tenant is member of within a given tenant

### PR DESCRIPTION
**Pull request by simcod**

### CODE REVIEW

**Review by Gerrit91 — COMMENTED** [review]
This looks pretty good to me now. I have two more questions though:

1. The PR title says you want toreturn the project annotations as well. Is this still planned? Because I think it's not happening at the moment.
2. In your test case, the Github tenant has a tenant membership with editor annotation, so implicitly he has access to all projects of the Github tenant. Why is there only project `1`  returned?

**Comment by Gerrit91** [comment]
@simcod Ping?

**Comment by simcod** [comment]
> @simcod Ping?

Sorry, did not had time yet. Will have a look at it in the next few days.

File: pkg/service/tenant_test.go [hunk 1]
```diff
@@ -797,6 +800,62 @@ func Test_tenantService_ListTenantMembers(t *testing.T) {
...
+				err = projectStore.Create(ctx, &v1.Project{Meta: &v1.Meta{Id: "2"}, TenantId: "github"})
+				require.NoError(t, err)
+				err = projectMemberStore.Create(ctx, &v1.ProjectMember{Meta: &v1.Meta{Annotations: map[string]string{"metalstack.cloud/project-role": "owner"}}, ProjectId: "1", TenantId: "github"})
+				require.NoError(t, err)
+				err = projectMemberStore.Create(ctx, &v1.ProjectMember{Meta: &v1.Meta{Annotations: map[string]string{"metalstack.cloud/project-role": "owner"}}, ProjectId: "2", TenantId: "github"})
+				require.NoError(t, err)
+				err = projectMemberStore.Create(ctx, &v1.ProjectMember{Meta: &v1.Meta{Annotations: map[string]string{"metalstack.cloud/project-role": "viewer"}}, ProjectId: "2", TenantId: "azure"})
+				require.NoError(t, err)
+				err = tenantMemberStore.Create(ctx, &v1.TenantMember{Meta: &v1.Meta{Annotations: map[string]string{"metalstack.cloud/tenant-role": "owner"}}, MemberId: "github", TenantId: "github"})
+				require.NoError(t, err)
```
Comment (Gerrit91) [inline pkg/service/tenant_test.go:825]: This project does not know about metalstack.cloud:

```suggestion
				err = projectMemberStore.Create(ctx, &v1.ProjectMember{Meta: &v1.Meta{Annotations: map[string]string{"project-role": "owner"}}, ProjectId: "1", TenantId: "github"})
				require.NoError(t, err)
				err = projectMemberStore.Create(ctx, &v1.ProjectMember{Meta: &v1.Meta{Annotations: map[string]string{"project-role": "owner"}}, ProjectId: "2", TenantId: "github"})
				require.NoError(t, err)
				err = projectMemberStore.Create(ctx, &v1.ProjectMember{Meta: &v1.Meta{Annotations: map[string]string{"project-role": "viewer"}}, ProjectId: "2", TenantId: "azure"})
				require.NoError(t, err)
				err = tenantMemberStore.Create(ctx, &v1.TenantMember{Meta: &v1.Meta{Annotations: map[string]string{"tenant-role": "owner"}}, MemberId: "github", TenantId: "github"})
				require.NoError(t, err)
```

File: proto/v1/tenant.proto [hunk 2]
```diff
@@ -59,6 +59,7 @@ message TenantWithMembershipAnnotations {
   Tenant tenant = 1;
   map<string, string> project_annotations = 2;
   map<string, string> tenant_annotations = 3;
+  repeated string project_ids = 4;
```
Comment (Gerrit91) [inline proto/v1/tenant.proto:62]:
```suggestion
  // ProjectIDs contains all project IDs of this tenant for which an explicit project membership exists
  repeated string project_ids = 4;
```

**Review by Gerrit91 — COMMENTED** [review]
Only nit-picking.

**Review by Gerrit91 — APPROVED** [review]

**Pull request merged**